### PR TITLE
Support graceful for uds

### PR DESCRIPTION
**Pull request by taoyuanyuan**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

**Comment by taoyuanyuan** [comment]
#1693

File: test/util/xprotocol_rpc.go [hunk 1]
```diff
@@ -70,7 +70,11 @@ func NewRPCClient(t *testing.T, id string, proto types.ProtocolName) *RPCClient
 
 func (c *RPCClient) connect(addr string, tlsMng types.TLSClientContextManager) error {
 	stopChan := make(chan struct{})
-	remoteAddr, _ := net.ResolveTCPAddr("tcp", addr)
+	var remoteAddr net.Addr
+	remoteAddr, err := net.ResolveTCPAddr("tcp", addr)
+	if err != nil {
+		remoteAddr, _ = net.ResolveUnixAddr("unix", addr)
```
Comment (CodingSinger) [inline test/util/xprotocol_rpc.go:76]: 这里的err为啥不处理呢？

File: test/util/genconfig.go [hunk 2]
```diff
@@ -117,12 +118,17 @@ func NewWeightedCluster(name string, hosts []*WeightHost) v2.Cluster {
 }
 
 func NewListener(name, addr string, chains []v2.FilterChain) v2.Listener {
+	nw := "tcp"
+	if _, err := net.ResolveTCPAddr("tcp", addr); err != nil {
```
Comment (CodingSinger) [inline test/util/genconfig.go:122]: “tcp"改为nw？

File: test/util/xprotocol_rpc.go (hunk 1 above)
Comment (taoyuanyuan) [reply test/util/xprotocol_rpc.go:76]: 测试代码，不会报错，有报错在Connect()也会报错

**Comment by CodingSinger** [comment]
LGTM

**Review by CodingSinger — APPROVED** [review]

File: test/integrate/transfer/transfer_test.go [hunk 3]
```diff
@@ -90,7 +90,8 @@ func TestTransfer(t *testing.T) {
 
 	tc := integrate.NewXTestCase(t, bolt.ProtocolName, util.NewRPCServer(t, appaddr, bolt.ProtocolName))
 
-	tc.ClientMeshAddr = "127.0.0.1:12101"
+	os.Remove("/tmp/client.sock")
+	tc.ClientMeshAddr = "/tmp/client.sock"
```
Comment (nejisama) [inline test/integrate/transfer/transfer_test.go:94]: 这里是不是应该用unix开头的？

Comment (taoyuanyuan) [reply test/integrate/transfer/transfer_test.go:94]: host的配置才需要，listener有专门的network type配置

**Review by nejisama — APPROVED** [review]
